hotwax/preorder
# Implemented: permission plugin and disable inventory config update (toggles) in catalog detail page if user has not common admin permission (#175)

### Related Issues


Closes #175

### Short Description and Why It's Useful

Implemented the following - 

- Added authorization folder having rules, action, and authorization logic (index.ts).
- Changes in store --> modules --> user for permission management
- Changed the login flow (Now getting user profile in login action itself instead of calling another action).
- Changes in userServices.ts removed some of the API call functions and added some.
- Changes in main.ts using the permission plugin
- Changes in the router as permission to routes.
- Changes in the menu component for filtering out pages which has permission to be displayed ( Changes in selectedIndex Logiic)
- Added permission check in inventory config toggles (reserveInventory and holdPreorderPhysicalInventory) only allow to COMMON ADMIN.

### Steps To Verify -
- The login flow
- The menu items are to be filtered out by checking permissions.
- The permission rules and actions
- Able to change ecomstore in settings
- The catalog detail page is working correctly as expected.
- The store state mutations
-  API call in userServices.ts

### Screenshots of Visual Changes before/after (If There Are Any)

![tttttt](https://github.com/hotwax/preorder/assets/67117461/f6e99f10-c570-4680-9494-a9cb91a0b74d)


**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change


### Contribution and Currently Important Rules Acceptance


- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)
